RFC-0039-cuda-support.md

```diff
@@ -0,0 +1,197 @@
+
+# [CUDA version support]
+
+**Authors:**
+* @atalman @malfet @tinglvv @nWEIdia @ptrblck
+
+
+## **Motivation**
+
+The proposal is to provide two main benefits
+
+- Aligned decision policy provides transparent decision making;
+- Incorporate decision points in the early release process to stagger CUDA migration from the RC release integration work.
+
+## **Proposed Policy and Process**
+
+### We should introduce new version of CUDA when
+
+- It enables new important GPU architecture (For example Blackwell with CUDA-12.8)
+- It brings significant performance improvements
+- It fixes significant correctness issues
+- It significantly reduces binary/memory footprint
+
+### We would deprecate version of CUDA when
+
+As soon as we introduce a new Experimental Version we should consider moving the previous Experimental Version to Stable, and decommission the previous Stable version. Typically we want to support 2-3 versions of CUDA as follows:
+
+- Optional Legacy Version: If we need to have 1 version for backend compatibility or to work around the current limitation. For example: CUDA older driver is incompatible with newer CUDA version
```

Comment:
> For example: CUDA older driver is incompatible with newer CUDA version

This is always the case for major updates in the CUDA toolkit. We kept CUDA 11.8 alive for the entire lifetime of CUDA 12.x as an unknown number of users supposedly could not update their NVIDIA driver in 2+ years (I don't know how many users). I would suggest keeping the "legacy" version as static as possible (i.e. no cuDNN, NCCL, etc. updates) to avoid mixing the legacy stack with latest libs and to reduce the churn in keeping the stable stack up-to-date.